plugins/inputs/procstat/nativeFinder_windows.go

```diff
@@ -0,0 +1,109 @@
+// +build windows
+
+package procstat
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/StackExchange/wmi"
+	"github.com/shirou/gopsutil/process"
+)
+
+//Timeout is the timeout used when making wmi calls
+var Timeout = 5 * time.Second
+
+func init() {
+	wmi.DefaultClient.AllowMissingFields = true
+}
+
+type queryType string
+
+const (
+	like     = queryType("LIKE")
+	equals   = queryType("=")
+	notEqual = queryType("!=")
+)
+
+// //Pattern matches the process name on windows and will find a pattern using a WMI like query
+// func (pg *NativeFinder) Pattern(pattern string) ([]PID, error) {
+// 	var pids []PID
+// 	procs, err := getWin32ProcsByVariable("Name", like, pattern, Timeout)
+// 	if err != nil {
+// 		return pids, err
+// 	}
+// 	for _, p := range procs {
+// 		pids = append(pids, PID(p.ProcessID))
+// 	}
+// 	return pids, nil
+// }
+
+//Pattern matches on the process name
+func (pg *NativeFinder) Pattern(pattern string) ([]PID, error) {
```

Comment:
I think this looks the same as the notwindows version, lets put shared implementation in a `native_finder.go` file.

Reply:
Agreed